phy/traces/detect.py

```diff
@@ -190,6 +194,16 @@ def connected_components(weak_crossings=None,
     if probe_adjacency_list is None:
         probe_adjacency_list = {}
 
+    if channels is None:
+        channels = []
+
+    # If the channels aren't referenced at all but exist in 'channels', add a
+    # trivial self-connection so temporal floodfill will work. If this channel
+    # is dead, it should be removed from 'channels'
+    for i in channels:
+        if not probe_adjacency_list.get(i):
+            probe_adjacency_list[i] = {i}
+
```

Comment:
`probe_adjacency_list = {i for i in channels if not probe_adjacency_list.get(i, None)}` ?